worker-languagedetection/src/main/java/com/hpe/caf/worker/languagedetection/LanguageDetectionWorker.java

```diff
@@ -153,11 +165,29 @@ private void detectLanguage(final Document document, final String fieldName, fin
             final LanguageDetectorResult detectorResult = languageDetector.detectLanguage(sequenceInputStream);
 
             if (detectorResult != null) {
-                addDetectedLanguageToDocument(detectorResult, document, sourceDataField, inMultiFieldMode);
+                addDetectedLanguageToDocument(detectorResult, document, sourceDataField, resultFormat, inMultiFieldMode);
             }
             //  Output response data (i.e. document field value changes).
             outputDocumentFieldValueChanges(document);
 
         }
     }
+
+    /**
+     * Determines the result output format that should be used with current document.
+     * @param document Document that results will be output for.
+     * @return the result format to use when outputting language detection results.
+     */
+    private LanguageDetectionResultFormat getResultFormatToUse(Document document)
+    {
+        final String resultFormatStr = document.getCustomData(LanguageDetectionConstants.CustomData.RESULT_FORMAT);
+        if(Strings.isNullOrEmpty(resultFormatStr)){
+            return LanguageDetectionResultFormat.SIMPLE;
+        }
+        else {
+            LanguageDetectionResultFormat convertedResultFormat =
+                    LanguageDetectionResultFormat.tryGetValueOf(resultFormatStr);
```

Comment:
This is similar to my previous comment about having tryGetValueOf - if there's an invalid value, we should fail. I looked into the language detection worker and there's quite a lot of bad exception handling so in this case we should: use the valueOf method and catch the Invalid argument exception thrown (it's unfortunate that it's a runtime exception but we should add in this case an error to the document).

Reply:
OK.